articles/architecture-scenarios/application/mobile-api.md

```diff
@@ -9,30 +9,369 @@ beta: true
+- __Client__: an application requesting access to a protected resource on behalf of the Resource Owner.
+- __Authorization Server__: the server that authenticates the Resource Owner, and issues access tokens after getting proper authorization. In this case, Auth0.
+
+Using [different grants types (or flows)](/api-auth/which-oauth-flow-to-use), these participants will interact to grant to the client apps limited access to the APIs you are building. As a result, the client app will obtain an `access_token` that can be used to call the API on behalf of the user.
+:::
+
+## Proof Key for Code Exchange (PKCE)
+
+OAuth 2 provides several grant types for different use cases. In this particular use case, we want to access the API from a mobile application, which will use the OAuth 2.0 [Proof Key for Code Exchange](/api-auth/grant/authorization-code-pkce) to do so.
+
+The [Authorization Code Grant](/api-auth/grant/authorization-code) has some security issues, when implemented on native applications. For instance, a malicious attacker can intercept the `authorization_code` returned by Auth0 and exchange it for an [Access Token](/tokens/access-token) (and possibly a [Refresh Token](/tokens/refresh-token)).
+
+The Proof Key for Code Exchange (PKCE) (defined in [RFC 7636](https://tools.ietf.org/html/rfc7636)) is a technique used to mitigate this authorization code interception attack.
+
+With PKCE, the Client creates, for every authorization request, a cryptographically random key called `code_verifier` and its transformed value called `code_challenge`, which is sent to Auth0 to get the `authorization_code`. When the Client receives the `authorization_code`, it will send the code and the `code_verifier` to Auth0's token endpoint to exchange them for the requested tokens.
+
+![PKCE](/media/articles/architecture-scenarios/mobile-api/authorization-code-grant-pkce.png)
+
+1. The native app initiates the flow and redirects the user to Auth0 (specifically to the [/authorize endpoint](/api/authentication#authorization-code-grant-pkce-)), sending the `code_challenge` and `code_challenge_method` parameters.
+2. Auth0 redirects the user to the native app with an `authorization_code` in the querystring.
+3. The native app sends the `authorization_code` and `code_verifier` together with the `redirect_uri` and the `client_id` to Auth0. This is done using the [/oauth/token endpoint](/api/authentication?http#authorization-code-pkce-).
+4. Auth0 validates this information and returns an `access_token` (and optionally a `refresh_token`).
+5. The native app can use the `access_token` to call the API on behalf of the user.
+
+## Auth0 Configuration
+
+In this section we will review all the configurations we need to apply at the [Auth0 Dashboard](https://manage.auth0.com/).
+
+### Create the API
+
+Click on the [APIs menu option](https://manage.auth0.com/#/apis) on the left, and click the Create API button.
```

Comment:
Do not hardcode the dashboard URL, use instead the variable `${manage_url}`

See https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#document-variables